src/Device/jprogOperations.ts

```diff
@@ -109,11 +114,11 @@ const verifySerialPortAvailable = device => {
  * @param {String} fwVersion The firmware version to be matched.
  * @returns {Promise} Promise that resolves if successful or rejects with error.
  */
-async function validateFirmware(device, fwVersion) {
-    let valid = false;
-    let fwInfo;
+export async function validateFirmware(device: Device, fwVersion: any) {
```

Comment:
TS type says `any` but jsdoc says `String`?

Reply:
Typescript doesn't like that we do a `typeof fwVersion.validator === 'function'`. String doesn't have a validator property, so jsdoc must be wrong.

Reply:
Oh right, this is dependent on the app. I think for BLE it's an object with a validator `property` plus maybe some other things, while for other apps it's a string. So it should probably be of type `{ validator: Function, ... } | string`